File: src/ColoredRect.java
import java.awt.*;

public class ColoredRect extends DrawableRect {
    protected Color inColor;

    public ColoredRect(int xx1, int yy1, int xx2, int yy2, Color rCol, Color inCol) {
        super(xx1, yy1, xx2, yy2, rCol);
        inColor = inCol;
    }

    public ColoredRect(int width, int height, Color rCol, Color inCol) {
        super(width, height, rCol);
        inColor = inCol;
    }

    public ColoredRect() {
        super();
        inColor = Color.blue;
    }

    @Override
    public void Draw(Graphics g) {
        g.setColor(super.outColor);
        g.fillRect(x1, y1, x2, y2);
        g.setColor(inColor);
        g.drawRect(x1, y1, x2, y2);
    }
}

File: src/ColoredRect.kt
import java.awt.*

class ColoredRect : DrawableRect {
    protected var inColor: Color

    constructor(xx1: Int, yy1: Int, xx2: Int, yy2: Int, rCol: Color, inCol: Color) : super(xx1, yy1, xx2, yy2, rCol) {
        inColor = inCol
    }

    constructor(width: Int, height: Int, rCol: Color, inCol: Color) : super(width, height, rCol) {
        inColor = inCol
    }

    constructor() : super() {
        inColor = Color.blue
    }

    override fun Draw(g: Graphics) {
        g.color = super.outColor
        g.fillRect(x1, y1, x2, y2)
        g.color = inColor
        g.drawRect(x1, y1, x2, y2)
    }
}
